BlueprintUI/Sources/Layout/Builder/StackChild.swift

```diff
@@ -0,0 +1,91 @@
+import UIKit
+
+/// `StackChild` is a wrapper which allows an element to define its StackLayout traits and Keys.
+/// This struct is particularly useful when working with `@ElementBuilder`.
+/// By default, elements will default to a nil key and the default `StackLayout.Traits` initializer.
+/// `@StackElementBuilder` will check every child to see if it can be type cast to a `StackChild`
+/// and then pull of the given traits and key and then apply those to the stack
+public struct StackChild {
+    public let element: Element
+    public var traits: StackLayout.Traits
+    public var key: AnyHashable?
+    
+    public struct Sizing {
+        public static let fixed: Self = .init(growPriority: 0, shrinkPriority: 0)
+        public static let flexible: Self = .init(growPriority: 1, shrinkPriority: 1)
+
+        public var growPriority: CGFloat
+        public var shrinkPriority: CGFloat
+        
+        public init(growPriority: CGFloat = 1, shrinkPriority: CGFloat = 1) {
+            self.growPriority = growPriority
+            self.shrinkPriority = shrinkPriority
+        }
+    }
+
+    public init(
+        element: Element,
+        traits: StackLayout.Traits = .init(),
+        key: AnyHashable? = nil
+    ) {
+        self.element = element
+        self.traits = traits
+        self.key = key
+    }
+    
+    public init(
+        element: Element,
+        sizing: Sizing = .flexible,
+        alignmentGuide: ((ElementDimensions) -> CGFloat)? = nil,
+        key: AnyHashable? = nil
+    ) {
+        self.init(
+            element: element,
+            traits: .init(
+                growPriority: sizing.growPriority,
+                shrinkPriority: sizing.shrinkPriority,
+                alignmentGuide: alignmentGuide.map(StackLayout.AlignmentGuide.init)
+            ),
+            key: key
+        )
+    }
+}
+
+extension Element {
+    
+    /// Wraps an element with a `StackChild` in order to customize `StackLayout.Traits` and the key.
+    /// - Parameters:
+    ///   - sizing: Controls the amount of extra space distributed to this child during underflow and overflow
+    ///   - alignmentGuide: Allows for custom alignment of a child along the cross axis.
+    ///   - key: A key used to disambiguate children between subsequent updates of the view
+    ///     hierarchy.
+    /// - Returns: A wrapped element with additional layout information for the `StackElement`.
+    public func stackChild(
```

Comment:
I'm going to bikeshed for `stackTraits` here. As a modifier, this adds traits to a child; every item without it is already a child.

Reply:
`stackItem`? e.g.:

```swift
Column {
    Label(...)
        .stackItem(sizing: .flexible)
}
```

I am kind of a fan of naming this after the type it is converted into, since it's in the global `Element` scope, but I don't feel too strongly about it.

Reply:
I'm thinking that stackTraits might get confused with StackLayout.Traits. Also, this struct is closer to a `child` since `children` in StackElement is defined as` [(element: Element, traits: StackLayout.Traits, key: AnyHashable?)]` which is pretty much the shape of StackChild. I would almost be tempted to change `children` from an array of tuples to `[StackChild]` but that would cause a lot of regressions.

Reply:
> I would almost be tempted to change `children` from an array of tuples to `[StackChild]` but that would cause a lot of regressions.

I'd encourage this — most stacks are built with the `add` methods, and don't access the `children` directly.

Reply:
Hm, even though it' not widely used, wouldn't this still be a breaking change? I could make a new `stackChildren: [StackLayout.Child]` (naming? _children? something else?) and then keep `children: [(element: Element, traits: StackLayout.Traits, key: AnyHashable?)]` as a computed variable that derives from [StackLayout.Child]. Then I could add a deprecation warning on top.

Reply:
It would be a breaking changes - but we (UIS) can handle applying this change.

Reply:
Cool! Then I'll go ahead and make it.